packages/unmock-core/src/__tests__/state.test.ts

```diff
@@ -0,0 +1,209 @@
+import { State } from "../service/state/state";
+
+const fullSchema = {
+  paths: {
+    "/test/{test_id}": {
+      get: {
+        parameters: [
+          {
+            name: "test_id",
+            in: "path",
+          },
+        ],
+        responses: {
+          200: {
+            description: "bar",
+            content: {
+              "application/json": {
+                schema: {
+                  type: "array",
+                  items: {
+                    properties: {
+                      foo: {
+                        type: "object",
+                        properties: {
+                          id: {
+                            type: "integer",
+                          },
+                        },
+                      },
+                      id: {
+                        type: "integer",
+                      },
+                    },
+                  },
+                },
+              },
+              "text/plain": {
+                schema: {
+                  properties: {
+                    notId: {
+                      type: "string",
+                    },
+                  },
+                },
+              },
+            },
+          },
+          default: {
+            description: "baz",
+            content: {
+              "application/json": {
+                schema: {
+                  properties: {
+                    error: {
+                      type: "string",
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+};
+
+describe("Test state management", () => {
+  it("returns undefined when setting empty state", () => {
+    const state = new State();
+    expect(
+      state.update({
+        stateInput: { method: "any", endpoint: "**", newState: {} },
+        serviceName: "foo",
+        paths: fullSchema.paths,
+        schemaEndpoint: "**",
+      }),
+    ).toBeUndefined();
+  });
+
+  it("returns error when setting state for non-existing method with generic endpoint", () => {
+    const state = new State();
+    expect(
+      state.update({
+        stateInput: { method: "post", endpoint: "**", newState: {} },
+        serviceName: "foo",
+        paths: fullSchema.paths,
+        schemaEndpoint: "**",
+      }),
+    ).toContain("Can't find any endpoints with method 'post'");
+  });
+
+  it("returns error when setting state for non-existing method with specific, existing endpoint", () => {
+    const state = new State();
+    expect(
+      state.update({
+        stateInput: { method: "post", endpoint: "/test/5", newState: {} },
+        serviceName: "foo",
+        paths: fullSchema.paths,
+        schemaEndpoint: "/test/{test_id}",
+      }),
+    ).toContain("Can't find response for 'post /test/5'");
+  });
+
+  it("saves state from any endpoint and get method as expected", () => {
```

Comment:
Irrespective of the implementation, it would be good to write a test that modifies the top-level `id` without modifying `foo/id` and vice-versa.

Reply:
Done! (removed the nested search as agreed)